std/string.d

```diff
@@ -4789,17 +4798,45 @@ body
  * </pre>
  */
 
-string[string] abbrev(string[] values) @safe pure
+S[S] abbrev(S)(S[] values)
+    if (isSomeString!S)
 {
     import std.algorithm : sort;
+    import std.utf : stride;
+    import core.stdc.stdlib;
 
-    string[string] result;
-
-    // Make a copy when sorting so we follow COW principles.
-    values = values.dup;
-    sort(values);
+    S[S] result;
 
+    // Make a copy so it can be sorted in place
     size_t values_length = values.length;
+
+    S* vptr;
+    S[] vsorted;
+    if (__ctfe)
+        vsorted = values.dup;
+    else
+    {
+        void alloc() @trusted nothrow @nogc
```

Comment:
This could be replaced by #3031 if/when that is merged (just cross-referencing so I can do the work then).